content/en/_index.md

```diff
@@ -1028,9 +1203,86 @@ def read_a_file file_path:
   puts "Contents: #{content}"
 end
 {{< /highlight >}}
+{{< /content-tab >}}
+{{< content-tab header="Terraform" >}}
+#### No CLIs
+
+Do not include CLI commands (such as gcloud or kubectl) inside
+of your sample via the `null_resource` resource.
+
+#### No Variables
+
+Don't include Terraform variables (`var.VARIABLE_NAME`). Variables
+enabling users to use Terraform samples not as snippets but as direct
+module references. Disabling the avenue of customization 
+reduces the risk so that users don't take accidental 
+dependency.
+Instead, hard code the resource names and attribute values.
+In some cases, [locals](#locals) are allowed.
+
+#### Locals {#locals}
+
+Sometimes multiple resources have a common string that is not
+accessible as a reference or there is no child/parent
+reference. Locals are allowed and encouraged for repeated text
+that a user may want to change. When a name is used multiple
+times, it's easier to copy, paste, and edit a local than search
+and replace the whole file. Locals are allowed and variable blocks
+are not because locals are an internal implementation and not
+part of the API  exposed via Terraform. Locals should be used only
+when necessary, and are best used when a repeated value is used
+throughout the configuration (3 or more times).
+
+#### Outputs
+
+Output statement provide values that are generated after creating
+a resource. They are allowed because they are readonly and they
+give  users some important information such as URLs and IP addresses.
+To keep it simple, they must be defined inline at the end of
+a  sample, not in a separate `outputs.tf` file.
+
+#### Resource references
+
+When possible, refer to already-defined resources.
+
+#### Terraform Google Provider version
```

Comment:
Provider version/argument/storage backend are things I mentioned above as maybe fitting into Client Initialization. Having them as broken out rules here seems fine too.

#StyleGuideFeedback

Reply:
Tracking in b/271024057, so closing here.